### ISSUE TITLE
Single Regression task on GraphConv model - training and evaluting did not work properly?

### ISSUE THREAD

**Issue by albertma-evotec** [deepchem/deepchem#1984]
**Description:**

I have been trying to run a single regression task (logD prediction) using the the GraphConv model. The problems I am seeing are the loss does not decrease during training. I am expecting an exponential decay-like curve. And also the r2 score I got is extremely poor when I evaluate the model even on the training set itself. I have tried different transformer and no transformer but I still see similar results. I am just wonder if the GraphConv.fit() and the evaluate() methods are known to have some issues?

May I ask if anyone has successfully run a regression task on the graphconv model?

![image](https://user-images.githubusercontent.com/54853041/86923580-2280b600-c126-11ea-9772-da8634ab9496.png)
![image](https://user-images.githubusercontent.com/54853041/86923591-26acd380-c126-11ea-86aa-61ac56a23913.png)
![image](https://user-images.githubusercontent.com/54853041/86923607-29a7c400-c126-11ea-9a57-e78e57a3039e.png)

- DeepChem Version: v2.3.0
- Platform:  linux

Below is the one of the examples in which I did not use a transformer.

```
dataset_file = 'LOGD_Canonical_SMILES.csv.gz'        #this file contains the canonical smiles and their logD value
task = ['VALUE_AS_NUMBER']                               #this is the y in the dataset. It is numerical and ranged from about -2 to 6
featurizer_func = dc.feat.ConvMolFeaturizer()
loader = dc.data.CSVLoader(tasks=task, smiles_field='Canonical_SMILES', featurizer=featurizer_func)
dataset = loader.featurize(dataset_file)

splitter = dc.splits.SpecifiedSplitter(dataset_file, "split", verbose=True)          #There is a "split" field in the dataset file where I  defined the training/valid/test set
train_dataset, valid_dataset, test_dataset = splitter.train_valid_test_split(dataset)

model = GraphConvModel(n_tasks=1, batch_size=100, mode='regression', dropout=0.25)
num_epochs = 20
losses = []
for i in range(num_epochs):
    loss = model.fit(train_dataset, nb_epoch=1)
    print("Epoch %d loss: %f" % (i, loss))
    losses.append(loss)
```

```
import matplotlib.pyplot as plot

#print(len(losses))

plot.ylabel("Loss")
plot.xlabel("Epoch")
x = range(num_epochs)
y = losses
plot.scatter(x, y)
plot.show()
```
![image](https://user-images.githubusercontent.com/54853041/86922643-dc772280-c124-11ea-97a3-273c85150898.png)
```
metric = dc.metrics.Metric(dc.metrics.pearson_r2_score, np.mean)
print("Evaluating model")
train_scores = model.evaluate(train_dataset, [metric])
valid_scores = model.evaluate(valid_dataset, [metric])
```
Evaluating model
computed_metrics: [0.06755689119181646]
computed_metrics: [0.07501060646752626]

Below is just to show what the true_y and the pred_y are
![image](https://user-images.githubusercontent.com/54853041/86924325-3a0c6e80-c127-11ea-8dee-dcfef95483f1.png)

**Comment by rbharath** [comment]
Which deepchem version are you on? I'd recommend upgrading to nightly. You can do this by running `pip install --pre deepchem`. I ran the following script:

```
import numpy as np

from deepchem.models import GraphConvModel
import matplotlib.pyplot as plt


np.random.seed(123)
import tensorflow as tf

tf.random.set_seed(123)
import deepchem as dc
from deepchem.molnet import load_delaney

# Load Delaney dataset
delaney_tasks, delaney_datasets, transformers = load_delaney(
    featurizer='GraphConv', split='index')
train_dataset, valid_dataset, test_dataset = delaney_datasets

# Fit models
metric = dc.metrics.Metric(dc.metrics.pearson_r2_score, np.mean)

# Do setup required for tf/keras models
# Number of features on conv-mols
n_feat = 75
# Batch size of models
batch_size = 128
model = GraphConvModel(
    len(delaney_tasks), batch_size=batch_size, mode='regression')

# Fit trained model
losses = []
for i in range(20):
  loss = model.fit(train_dataset, nb_epoch=1)
  print("loss: %s" % str(loss))
  losses.append(loss)
print("losses")
print(losses)

print("Evaluating model")
train_scores = model.evaluate(train_dataset, [metric], transformers)
valid_scores = model.evaluate(valid_dataset, [metric], transformers)

print("Train scores")
print(train_scores)

print("Validation scores")
print(valid_scores)

plt.plot(losses)
plt.show()
```

And I got a very smooth set of losses. Can you give this a try?

**Comment by albertma-evotec** [comment]
my deepchem version is v.2.3.0. Do i need to uninstall the original deepchem after pip install --pre deepchem ??
what is the difference between the deepchem-nightly and deepchem modules? 
Both of them are v2.3.0
![image](https://user-images.githubusercontent.com/54853041/87024499-1e5da280-c1d1-11ea-87a8-07aec22bf0fe.png)

**Comment by nd-02110114** [comment]
@albertma-evotec 
we are so sorry for the confusion about the installation.

Currently, `pip install --pre deepchem` is correct.
So, I recommend you to reinstall the deepchem.

```
pip uninstall deepchem deepchem-nightly
pip install --pre deepchem
```

**Comment by albertma-evotec** [comment]
It seems working now, thanks!
Although I found some strange phenomena, the loss periodically dropped to zero for some reasons.

![image](https://user-images.githubusercontent.com/54853041/87455248-0f298b00-c5fd-11ea-979d-79a77a81ede8.png)
![image](https://user-images.githubusercontent.com/54853041/87455263-13ee3f00-c5fd-11ea-956f-e1e44bc90610.png)
![image](https://user-images.githubusercontent.com/54853041/87455281-19e42000-c5fd-11ea-86f5-074b561adc26.png)

**Comment by rbharath** [comment]
@albertma-evotec The loss dropping to 0 is a bug in our logging code (https://github.com/deepchem/deepchem/issues/1944). It's harmless for the actual model training. Glad to see the model is working!
